### PR TITLE
Lumine Hall's VWF

### PR DESCRIPTION
**Pull request by Lorenzooone**
Fixes https://github.com/jeffman/Mother2GbaTranslation/issues/98.
Fixes https://github.com/jeffman/Mother2GbaTranslation/issues/99.

### CODE REVIEW

File: src/c/luminehall.c [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    int tileHeight = 2;
+    int chosenLen;
+    int renderedLen;
+    byte *glyphRows;
+    int AlternativeTiles[4];
+    switch(chr)
+    {
+        case END:
+            return;
+        case ENDL:
```
Comment (PhoenixBound) [inline src/c/luminehall.c:56]: Was there a specific reason for making an "ENDL" code? The original game just used [two spaces](https://github.com/Herringway/ebsrc/blob/8151d9b9521294f18d6da82846ceecc6f675f3bf/src/data/text/lumine_hall.asm) for that part, it looks like.

File: src/data/lumine-table.tbl [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+5C=|
+5D=}
+5E=~
+5F=[AF]
+
+F9=\cness
+FA=\cpaula
+FB=\cjeff
+FC=\cpoo
+/00
```
Comment (PhoenixBound) [inline src/data/lumine-table.tbl:117]: This part changes the byte used for "end" of a string defined with `.str`. You can just set it as /FF and not need to use the \end manually in the actual text.

File: src/data/lumine-text.asm [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.loadtable "data/lumine-table.tbl"
+
+.strn   "\blank"
+.strn   "I'm \cness....\nIt's been a long road getting here...\n"
```
Comment (PhoenixBound) [inline src/data/lumine-text.asm:4]: Counterintuitive but for some reason the game never actually prints the first period after Ness's name. Just `I'm Ness...  It's` etc

File: src/m2-hack.asm [hunk 4]
```diff
@@ -1822,6 +1818,12 @@ m2_nybbles_to_bits:
 m2_enemy_attributes:
 .incbin "data/m2-enemy-attributes.bin"
 
+luminesquaretable:
```
Comment (PhoenixBound) [inline src/m2-hack.asm:1821]: `.align 2`

File: src/data/lumine-text.asm (hunk 3 above)
Comment (Lorenzooone) [reply src/data/lumine-text.asm:4]: You sure about this one?

File: src/c/luminehall.c (hunk 1 above)
Comment (Lorenzooone) [reply src/c/luminehall.c:56]: Thought it was 8 pixels instead of 6. Fixing it.

File: src/data/lumine-text.asm (hunk 3 above)
Comment (Lorenzooone) [reply src/data/lumine-text.asm:4]: https://www.youtube.com/watch?v=cFsibAm0Iy8

File: src/data/lumine-table.tbl (hunk 2 above)
Comment (Lorenzooone) [reply src/data/lumine-table.tbl:117]: Cool!

File: src/c/luminehall.c [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+    setTilesToBlank(Tiles);
+    //First things first, it calculates the total length of the text in pixels
+    while((*lumineText_curr_ptr) != END)
+    {
+        length += getCharWidth(*lumineText_curr_ptr);
+        lumineText_curr_ptr++;
+    }
+    //It then gets the length in arrangements. It also makes sure it's a number divisible by 8.
+    //This avoids having empty tiles showing
+    length = (length & 8 == 0 ? length : (length + (8 - (length & 7)))) >> 1;
```
[the remark this answers is not in the document]
Comment (Lorenzooone) [reply src/c/luminehall.c]: Fixed.

**Pull request merged**